crates/holochain_state/src/source_chain.rs

```diff
@@ -1891,4 +1901,127 @@ pub mod tests {
+            ChainQueryFilterRange::Unbounded,
+            ChainQueryFilterRange::ActionSeqRange(0, 2),
+            ChainQueryFilterRange::ActionHashRange(
+                elements[0].action_address().clone(),
+                elements[2].action_address().clone(),
+            ),
+            ChainQueryFilterRange::ActionHashTerminated(elements[2].action_address().clone(), 2),
+        ];
+
+        // A variety of combinations of query parameters
+        let cases = [
+            ((None, None, vec![], false), 3),
+            ((None, None, vec![], true), 3),
+            ((Some(ActionType::Dna), None, vec![], false), 1),
+            ((None, Some(EntryType::AgentPubKey), vec![], false), 1),
+            ((None, Some(EntryType::AgentPubKey), vec![], true), 1),
+            ((Some(ActionType::Create), None, vec![], false), 1),
+            ((Some(ActionType::Create), None, vec![], true), 1),
+            (
+                (
+                    Some(ActionType::Create),
+                    Some(EntryType::AgentPubKey),
+                    vec![],
+                    false,
+                ),
+                1,
+            ),
+            (
+                (
+                    Some(ActionType::Create),
+                    Some(EntryType::AgentPubKey),
+                    vec![elements[2].action().entry_hash().unwrap().clone()],
+                    true,
+                ),
+                1,
+            ),
+        ];
+
+        // Test all permutations of cases defined with all full range queries,
+        // and both boolean values of `include_entries`.
+        for ((action_type, entry_type, entry_hashes, include_entries), num_expected) in cases {
+            let entry_hashes = if entry_hashes.is_empty() {
+                None
+            } else {
+                Some(entry_hashes.into_iter().collect())
+            };
+            for sequence_range in full_ranges.clone() {
+                let query = ChainQueryFilter {
+                    sequence_range: sequence_range.clone(),
+                    action_type: action_type.clone(),
+                    entry_type: entry_type.clone(),
+                    entry_hashes: entry_hashes.clone(),
+                    include_entries,
+                };
+                if sequence_range != ChainQueryFilterRange::Unbounded
+                    && (action_type.is_some()
+                        || entry_type.is_some()
+                        || entry_hashes.is_some()
+                        || include_entries)
+                {
+                    assert!(matches!(
+                        chain.query(query.clone()).await,
+                        Err(SourceChainError::UnsupportedQuery(_))
+                    ));
+                } else {
+                    let queried = chain.query(query.clone()).await.unwrap();
+                    let actual = queried.len();
+                    assert!(queried.iter().all(|e| e.action().author() == &alice));
```

Comment:
where are the entries created? i'm trying to verify that there are entries by alice **and** bob to begin with

Reply:
The entries are created by `genesis` above